ui/lib/apps/InstanceProfiling/pages/Detail.tsx

```diff
@@ -87,56 +98,11 @@ async function getActionToken(
   return token
 }
 
-function ViewResultButton({ rec, t }) {
-  const openResult = usePersistFn(async (openAs: string) => {
-    const isProtobuf = rec.raw_data_type === RawDataType.Protobuf
-    let token: string | undefined
-    let profileURL: string
-
-    switch (openAs) {
-      case ViewOption.Download:
-        token = await getActionToken(rec.id, 'single_download')
-        if (!token) {
-          return
-        }
-
-        window.location.href = `${client.getBasePath()}/profiling/single/download?token=${token}`
-        break
-      case ViewOption.FlameGraph:
-        token = await getActionToken(rec.id, 'single_view')
-        if (!token) {
-          return
-        }
-
-        profileURL = `${client.getBasePath()}/profiling/single/view?token=${token}`
-        if (isProtobuf) {
-          const titleOnTab = rec.target.kind + '_' + rec.target.display_name
-          profileURL = `/dashboard/speedscope#profileURL=${encodeURIComponent(
-            // protobuf can be rendered to flamegraph by speedscope
-            profileURL + `&output_type=protobuf`
-          )}&title=${titleOnTab}`
-        }
-
-        window.open(`${profileURL}`, '_blank')
-        break
-      case ViewOption.Graph:
-        token = await getActionToken(rec.id, 'single_view')
-        if (!token) {
-          return
-        }
-
-        profileURL =
-          profileURL = `${client.getBasePath()}/profiling/single/view?token=${token}&output_type=${
-            ViewOption.Graph
-          }`
-
-        window.open(`${profileURL}`, '_blank')
-    }
-  })
-
+function DropdownButton(props) {
+  const { rec, t } = props
```

Comment:
You can use the hook instead of passing `t` around in a component:

```suggestion
function DropdownButton({ rec }) {
  const { t } = useTranslation()
```